pkg/controllers/edgeconnect/controller.go

```diff
@@ -82,26 +82,42 @@ func (controller *Controller) SetupWithManager(mgr ctrl.Manager) error {
 }
 
 func (controller *Controller) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
-	log.Info("reconciling EdgeConnect", "name", request.Name, "namespace", request.Namespace)
+	llog := log.WithValues("namespace", request.Namespace, "name", request.Name)
+
+	logExit := llog.DebugLogFunctionBoundaries()
+	defer logExit()
+
+	llog.Info("reconciling EdgeConnect")
 
 	edgeConnect, err := controller.getEdgeConnect(ctx, request.Name, request.Namespace)
 	if err != nil {
-		log.Error(errors.WithStack(err), "reconciliation of EdgeConnect failed", "name", request.Name, "namespace", request.Namespace)
+		llog.Debug("reconciliation of EdgeConnect failed")
 
 		return reconcile.Result{}, err
 	} else if edgeConnect == nil {
+		llog.Debug("EdgeConnect object does not exist")
+
 		return reconcile.Result{}, nil
 	}
 
-	if edgeConnect.GetDeletionTimestamp() != nil {
+	if deletionTimestamp := edgeConnect.GetDeletionTimestamp(); deletionTimestamp != nil {
+		llog.Debug("EdgeConnect object shall be deleted", "timestamp", deletionTimestamp.String())
+
 		return reconcile.Result{}, controller.reconcileEdgeConnectDeletion(ctx, edgeConnect)
 	}
 
+	llog.Debug("EdgeConnect object needs reconcile")
+
 	return controller.reconcileEdgeConnect(ctx, edgeConnect)
 }
 
 func (controller *Controller) reconcileEdgeConnectDeletion(ctx context.Context, edgeConnect *edgeconnectv1alpha1.EdgeConnect) error {
-	log.Info("reconciling EdgeConnect deletion", "name", edgeConnect.Name, "namespace", edgeConnect.Namespace)
+	llog := log.WithValues("namespace", edgeConnect.Namespace, "name", edgeConnect.Name)
```

Comment:
Sidenote: now I kinda get why people like store the logger in the `ctx`, so they don't have to do this in the beginning of every function

Reply:
you still have to have an additional line to retrieve the logger from the context then (or add that to each log call). And sometimes you might want to add additional k/vs in a function. I don't think there's much won when adding the logger to the context.

Reply:
I am not saying we should do it, I am just starting to see a usecase for it :D